### DISCUSSION TITLE
PCAP not retreiving 443 traffic

### DISCUSSION THREAD

**Discussion by BradMic** [Security-Onion-Solutions/securityonion#10761] [General]
I get alerts about traffic on port 443 but it never returns anything when I select Actions > PCAP. For traffic on other ports Actions > PCAP works just fine. Although this really feels like a BPF, I do not have any BPFs configured for port 443 in the steno stanza of the global or minion salt pillars. Any ideas?

![2013933](https://github.com/Security-Onion-Solutions/securityonion/assets/119625312/3d6f0afd-406d-4355-ba04-285061b0cf9e)
![2013933_pcap](https://github.com/Security-Onion-Solutions/securityonion/assets/119625312/8d058b51-2716-4044-95c6-3a1f6bbd6734)

**Comment by cm-ops** [comment]
If you open one of those up what is the traffic?

**Reply by BradMic** [reply]
Thanks for replying. The second of the two pictures I posted shows what happens when I open one of them up. It says "No data available".

**Reply by cm-ops** [reply]
I was talking about opening up one in the top picture, I wanted to see what the network metadata for one of those logs is. There are certain protocols, like ICMP, that will not return a PCAP if you try to retrieve it.

**Reply by BradMic** [reply]
The expanded alert from the top pic shows it is TCP traffic from port 59686 to port 443. Sorry, can't show entire expanded metadata because it contains too much sensitive data.  ICMP doesn't use port numbers and I am able to retrieve ICMP traffic via PCAP if it matches my search criteria.

**Reply by cm-ops** [reply]
Have you tried to pivot to Hunt on one of those and correlate to see the Zeek logs that go with the alert?

**Reply by BradMic** [reply]
I can pivot to Hunt for one of the alerts on port 443 and correlate to see the Zeek logs that go with that alert but I don't see how that fixes my issue of Steno not saving any port 443 traffic.

**Reply by cm-ops** [reply]
If you try an pull the PCAP for the Zeek connection or HTTP log, are you able? If not, what is your disk space and how much PCAP retention are you seeing in Grafana?

**Reply by BradMic** [reply]
If the traffic is on port 443 then no. I can't pull PCAP from Alerts, Zeek or anything. If the traffic is not on port 443 then I have no problem pulling the PCAP. "I've pulled DNS HTTP, DHCP, RDP, STUN, etc. Disk space is not an issue. The disk is only using 29% of its capacity plus I have no issues pulling PCAP that is not on port 443.
![Screenshot 2023-07-19 095927](https://github.com/Security-Onion-Solutions/securityonion/assets/119625312/93711b43-3c22-405c-8973-91fb8d280f13)

**Reply by cm-ops** [reply]
Have you tried to pull a PCAP for port 443 from the CLI? If so, was the result the same? https://docs.securityonion.net/en/2.3/stenographer.html?highlight=so-pcap-export#command-line

**Reply by BradMic** [reply]
Yes, there are no packets in the output.

**Reply by BradMic** [reply]
It so much feels like a BPF but I removed the TLS BPF section from the steno stanza in /opt/so/saltstack/local/pillar/global.sls. It used to contain the following (markup messes up the spacing):
steno:
  bpf:
    - not tcp port 443 &&
    - not udp port 443 &&
    - not tcp port 8443 &&
    - not udp port 8443 &&

and then I ran _sudo salt "sensor-name' cmd.run 'so-pcap-restart'_ 

There are no BPFs in any of the minion files.

**Reply by cm-ops** [reply]
It does, check `cat /opt/so/conf/steno/config` it should be default like below:

```
{
  "Threads": [
    { "PacketsDirectory": "/nsm/pcap", "IndexDirectory": "/nsm/pcapindex", "MaxDirectoryFiles": 30000, "DiskFreePercentage": 10 }
  ]
  , "StenotypePath": "/usr/bin/stenotype"
  , "Interface": "bond0"
  , "Port": 1234
  , "Host": "127.0.0.1"
  , "Flags": ["-v", "--blocks=2048", "--preallocate_file_mb=4096", "--aiops=128", "--uid=stenographer", "--gid=stenographer"]
  , "CertPath": "/etc/stenographer/certs"
}
```

**Reply by BradMic** [reply]
On the sensor, the /opt/so/conf/steno/config file looks just like that except for the following differences:
MaxDirectoryFiles is set to 150000
The Flags line includes a --filter= option followed by a long string of hexadecimal numbers.

**Reply by cm-ops** [reply]
Do you have other steno BPFs enabled?

**Reply by BradMic** [reply]
yes

**Reply by BradMic** [reply]
I deleted the entire --filter option in /opt/so/conf/steno/config and ran _salt "sensor_name" state.highstate_

**Reply by cm-ops** [reply]
The BPF should be set in the pillar file on the manager (global or minion depending on where you set it). Then when the pcap state runs it will put the BPF in the config based on what is set there. If you remove from /opt/so/conf, salt will place the BPF back when the state is run.
